src/main/java/com/synopsys/integration/detect/tool/cache/InstalledToolLocator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * synopsys-detect
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.detect.tool.cache;
+
+import java.io.File;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+public class InstalledToolLocator {
+    private InstalledToolData installedToolData;
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    public InstalledToolLocator(Path pathToInstalledToolDataFile, Gson gson) {
+        File installedToolDataFile = pathToInstalledToolDataFile.toFile();
+        try {
+            if (!installedToolDataFile.getName().equals(InstalledToolManager.INSTALLED_TOOL_FILE_NAME)) {
+                // Path is to parent of cached inspector file
```

Comment:
I would probably check that installedToolDataFile is a directory rather than the name.
If you decide to continue checking the name that's fine, but you should verify it's a directory before constructing the new `File` object.